nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py

```diff
@@ -407,12 +407,17 @@ def forward(
             decoder_cross_attention_relative_position_bias,
         ) = (None, None, None)
 
-        enc_output_provided = enc_output is not None
-
         if enc_input is not None:
             # If enc_input is provided, we need to transpose it from [B x S x H] -> [S x B x H].
             enc_input = enc_input.transpose(0, 1)
-        elif (enc_input is None) and (enc_input_ids is not None):
+            enc_seq_length = enc_input.size(0)
+        elif enc_output is not None:
+            # If enc_output is provided in `batch_for_pipeline`, we need to transpose it from [B x S x H] -> [S x B x H].
+            enc_output = enc_output.transpose(0, 1)
+            enc_seq_length = enc_output.size(0)
+        # Only need to run encoder embedding and position ids if enc_input or enc_output is not provided.
+        else:
```

Comment:
I wonder if this if-else is what we intended?
- if 'enc_input' is not None
- elif 'enc_output' is not None
- else ...

Would this be an edge case are not covered with this change?
- if both 'enc_input' and 'enc_output' are not None at the same time

Reply:
Fixed, Thanks!